From 3d0a9b266b6941d77d3b314ad95527c23c7eb3f3 Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Wed, 17 Apr 2013 16:02:09 +0900
Subject: [PATCH 2/2] ext/dfbvideosink: recreate surface after changing the
 pixel format

This change is a temporal workaround to reflect the changing the
pixel format in the surface.
---
 ext/directfb/dfbvideosink.c |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/ext/directfb/dfbvideosink.c b/ext/directfb/dfbvideosink.c
index c2a4149..3e46fd9 100644
--- a/ext/directfb/dfbvideosink.c
+++ b/ext/directfb/dfbvideosink.c
@@ -1971,6 +1971,13 @@ gst_dfbvideosink_setcaps (GstBaseSink * bsink, GstCaps * caps)
       GST_DEBUG_OBJECT (dfbvideosink, "layer %d now configured to %dx%d %s",
           dfbvideosink->layer_id, lc.width, lc.height,
           gst_dfbvideosink_get_format_name (lc.pixelformat));
+
+      /* workaround: recreate the surface after changing the color. */
+      if (dfbvideosink->primary) {
+        dfbvideosink->primary->Release (dfbvideosink->primary);
+        dfbvideosink->layer->GetSurface (dfbvideosink->layer,
+            &dfbvideosink->primary);
+      }
     }
   }
 #if defined(HAVE_SHVIO)
-- 
1.7.5.4

